src/js/TreeNode.js

```diff
@@ -79,14 +76,13 @@ class TreeNode extends React.Component {
         }
 
         // Auto expand if enabled
-        if (this.hasChildren() && this.props.expandOnClick) {
+        if (!this.props.isLeaf && this.props.expandOnClick) {
             this.onExpand();
         }
 
         this.props.onClick({
             value: this.props.value,
             checked: isChecked,
-            children: this.props.rawChildren,
```

Comment:
This may have been used by some people. I personally do not use this, but to maintain backwards compatibility, I would like for the same information to be available.

Reply:
That makes perfect sense for compatibility. I'll add it back.

For any one else reading this, the children are also available in the `flatnodes` array through the `flatnodes.self.children`.

Reply:
I answered too quick. This object being passed to the `this.props.onclick` function sent from CheckTree is not really a node even though it is referred to as one in the `CheckTree.onClick` method. This caused me considerable confusion at first  :) thus the comments I added in various functions to keep track. I removed the `children` property because it was not necessary in changes being made for the Radio Groups #99 . There the original node was to be returned. This change should have been made here in the Refactor request to make this clear.

The original method from CheckTree:
```
  onCheck(node) {
        // node is object from TreeNode
        const { noCascade, onCheck } = this.props;

        this.toggleChecked(node, node.checked, noCascade);
        onCheck(this.serializeList('checked'), node);
    }
```

Here are the relevant parts of this function from CheckTree as used in the Radio Groups pull request. 
```
onCheck(nodeInfo) {
        // nodeInfo is object from TreeNode
        const { value, checked } = nodeInfo;

        const flatNode = this.flatNodes[value];
        const { parent, self } = flatNode;

        let { noCascade } = this.props;

        this.toggleChecked(self, checked, noCascade);
        this.props.onCheck(this.serializeList('checked'), self);
    }
```
So what gets sent out of CheckTree is the original node (`self`) sent into CheckTree and not the object sent from TreeNode.  Doing it this way, the children are already available without the need to add them in in TreeNode.

What may be missing is the value `checked` which is not in `self` but could be added for backwards compatibility.

Looking further, `onClick` should probably be handled the same way.

What do you think?